lib/Rx/Operator/ReduceOperator.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+
+namespace Rx\Operator;
+
+
+use Rx\ObservableInterface;
+use Rx\Observer\CallbackObserver;
+use Rx\ObserverInterface;
+use Rx\SchedulerInterface;
+
+
+class ReduceOperator implements OperatorInterface
+{
+
+    /** @var  callable */
+    protected $accumulator;
+    protected $seed;
+    protected $hasSeed;
+
+    function __construct($accumulator, $seed)
+    {
+        if (!is_callable($accumulator)) {
+            throw new \InvalidArgumentException('Accumulator should be a callable.');
+        }
+
+        $this->accumulator = $accumulator;
+        $this->seed        = $seed;
+        $this->hasSeed     = $seed ? true : false;
+    }
+
+
+    /**
+     * @param \Rx\ObservableInterface $observable
+     * @param \Rx\ObserverInterface $observer
+     * @param \Rx\SchedulerInterface $scheduler
+     * @return \Rx\DisposableInterface
+     */
+    public function __invoke(ObservableInterface $observable, ObserverInterface $observer, SchedulerInterface $scheduler = null)
+    {
+        $hasAccumulation = false;
+        $accumulation    = null;
+        $hasValue        = false;
+
+        return $observable->subscribe(new CallbackObserver(
+            function ($x) use ($observer, &$hasAccumulation, &$accumulation, &$hasValue) {
+
+                !$hasValue && ($hasValue = true);
```

Comment:
Just `$hasValue = true`? No need to check if it's false? Or am I missing something.